### ISSUE TITLE
TqdmSynchronisationWarning: Set changed size during iteration (see https://github.com/tqdm/tqdm/issues/481)

### ISSUE THREAD

**Issue by cni-md** [datalad/datalad#2468]
```
% datalad aggregate-metadata
[INFO   ] Aggregate metadata for dataset /home/jstadler/LIN/Projekte/3T/dltest 
Metadata extraction:  33%|████████████████████████████████████████                                                                                | 1.00/3.00 [00:00<00:00, 102k extractors/s]/home/jstadler/LIN/datalad/lib/python3.5/site-packages/tqdm/_monitor.py:89: TqdmSynchronisationWarning: Set changed size during iteration (see https://github.com/tqdm/tqdm/issues/481)
  TqdmSynchronisationWarning)
Metadata extraction:  67%|████████████████████████████████████████████████████████████████████████████████                                        | 2.00/3.00 [01:07<00:20, 20.3s/ extractors]/home/jstadler/LIN/datalad/lib/python3.5/site-packages/tqdm/_monitor.py:89: TqdmSynchronisationWarning: Set changed size during iteration (see https://github.com/tqdm/tqdm/issues/481)       
  TqdmSynchronisationWarning)
```

**Comment by mih** [comment]
Observation made during Brainhack.

**Comment by mih** [comment]
Glancing over peoples shoulders I have seen this on Linux boxes and Macs, so it seems to be a more common issue, but I have seen it first the first time that day.

**Comment by yarikoptic** [comment]
I think I saw it happen to me yesterday... May be I will manage you reproduce to fix it

**Comment by yarikoptic** [comment]
ah -- mine was different, for which I have pushed "fix" in #2492 ... here I wonder if we should just [suppress that warning ](https://github.com/tqdm/tqdm/issues/481#issuecomment-386603914) although may be it is indicative of our code needing a fix?

**Comment by mih** [comment]
I think this has been dealt with.
